ash/src/vk/extensions.rs

```diff
@@ -17672,6 +17672,20 @@ impl ExtPipelinePropertiesFn {
     pub const SPEC_VERSION: u32 = 1u32;
 }
 #[allow(non_camel_case_types)]
+#[doc = "Implemented for all types that can be passed as argument to `p_pipeline_properties` in [`PFN_vkGetPipelinePropertiesEXT`]"]
+pub unsafe trait PFN_vkGetPipelinePropertiesEXT_param_p_pipeline_properties_validstructs {
+    unsafe fn as_base_in_struct(&self) -> *const BaseInStructure {
+        <*const _>::cast(self)
+    }
+    unsafe fn as_base_out_struct(&mut self) -> *mut BaseOutStructure {
+        <*mut _>::cast(self)
+    }
```

Comment:
Perhaps we should just cast within the trait, but this seems to be more clearly conveying the use-case. Note that the original `extern "system" fn PFN_vkGetPipelinePropertiesEXT` doesn't use the trait but retains `vk.xml`'s `*mut vk::BaseOutStructure` type.

Reply:
I'd favor casting in the wrapper rather than introducing trait methods, just for the sake of simplicity.